dbflute-runtime/src/main/java/org/dbflute/cbean/ckey/ConditionKey.java

```diff
@@ -449,10 +450,19 @@ protected ColumnRealName resolveCompoundColumn(ColumnRealName baseRealName, Cond
         }
         final List<SpecifiedColumn> compoundColumnList = option.getCompoundColumnList();
         final List<ColumnRealName> realNameList = new ArrayList<ColumnRealName>();
-        realNameList.add(baseRealName); // already cipher
+        if (option.isNullCompoundedAsEmpty()) {
+            realNameList.add(toColumnRealName("coalesce(" + baseRealName + ",\'\')"));            
+        } else {
+            realNameList.add(baseRealName); // already cipher            
+        }
         for (SpecifiedColumn specifiedColumn : compoundColumnList) {
-            realNameList.add(doResolveCompoundColumn(option, specifiedColumn));
+            if (option.isNullCompoundedAsEmpty()) {
+                realNameList.add(toColumnRealName("coalesce(" + doResolveCompoundColumn(option, specifiedColumn) + ",\'\')"));
+            } else {
+                realNameList.add(doResolveCompoundColumn(option, specifiedColumn));
+            }
```

Comment:
Thank you.
But maybe you can refactor more like this:
```
realNameList.add(xxxxxx(option, baseRealName));
for (SpecifiedColumn specifiedColumn : compoundColumnList) {
    realNameList.add(xxxxxx(option, doResolveCompoundColumn(option, specifiedColumn))));
}
```

Reply:
> Thank you.
> But maybe you can refactor more like this:
> 
> realNameList.add(xxxxxx(option, baseRealName));
> for (SpecifiedColumn specifiedColumn : compoundColumnList) {
>     realNameList.add(xxxxxx(option, doResolveCompoundColumn(option, specifiedColumn))));
> }

refactored the codes as you suggested
https://github.com/dbflute/dbflute-core/pull/54/commits/37bf3a4b02b2c002b6779dc0b16542f0d8618c4e